dcos_launch/terraform.py

```diff
@@ -0,0 +1,144 @@
+import json
+import os
+import re
+import shutil
+import subprocess
+
+import dcos_launch.config
+import yaml
+from dcos_launch import gcp, util
+
+
+class TerraformLauncher(util.AbstractLauncher):
+    def __init__(self, config: dict, env=None):
+        self.config = config
+        self.init_dir = dcos_launch.config.expand_path('', self.config['init_dir'])
+        self.cluster_profile_path = os.path.join(self.init_dir, 'desired_cluster_profile.tfvars')
+
+        if config['platform'] == 'azure':
+            # TODO
+            pass
+        elif config['platform'] == 'aws':
+            # TODO
+            pass
+        elif config['platform'] == 'gcp':
+            creds_string, creds_path = gcp.get_credentials(env)
+            if not creds_path:
+                creds_path = os.path.abspath('gcp_service_creds.json')
```

Comment:
not a huge fan of copying creds from env and leaving them in files as plaintext. Given that you do this for every init, could we create this with this method? https://github.com/dcos/dcos-test-utils/blob/master/dcos_test_utils/helpers.py#L244

That way we will delete these files after most runs

Reply:
great idea, I'll use this both for creds and ssh keys

Reply:
Done